### ISSUE TITLE
Configurable routes

### ISSUE THREAD

**Issue by felorhik** [Riari/laravel-forum-frontend#7]
I brought this up in https://github.com/Riari/laravel-forum/issues/111

However I do find myself wanting to use the default frontend but heavily customize it. Without being able to directly manipulate the routes though it makes it hard to do it in an intuitive way.

What I recommend is a function similar to `Route::auth()` for the forum routes and some config variable to be able disable auto loading routes. This will give you full control over the routes, to completely remake them, or include them but add additional.

In my use case I am moving the category create functionality into its own view, but it needs a route. So sure I can add a route for it but it will be there alone, ungrouped with other similar routes. So when looking at it it can be difficult to understand the structure. If there is a function like `Forum::routes()` then you know that the routes are coming from there.

Anyway just my thoughts ^^ So far liking this but have run into some situations where I have been pulling out controllers and such as I need to modify them, don't want to do it for the whole package if I can help it :P (this is only the case with the frontend package)

**Comment by Riari** [comment]
I like the idea, I'm just not sure exactly how to implement it yet. I feel like the `middleware` config option should be removed at the same time to keep things simple, but that's not BC-friendly. There's also #6 to consider, which makes me think that `Forum::routes()` should only register the routes that exist immediately inside the main group that applies the `/forum` prefix, enabling people to define that group themselves if they disable the automatic loading.

Any thoughts on that?

**Comment by felorhik** [comment]
You could do something like this.

```
Forum::routes(function($route, $controllers) {
//apply custom routes under the special prefix
});
```

Basically the callback function can be optional to add additional routes under the same prefix, while also maintaining the regular middleware settings. Or if you put it under a route group the middleware settings are no longer necessary. If under another route group that has another prefix, the prefix in the settings can be set to null to disable it.

Another setting can be made to likely tell the `Forum::routes` function to do almost nothing other then pass the controllers, this way all routes can be customized depending on the level of need.

I see you intend for this package to be able to be used regardless of what it is used in. I find myself pulling controllers and policies out to do get what I need done but I can see that is part of the intent. However because of how the routes are more or less forced if you use the service provider I can see it could be beneficial to set up the ability to customize them or even disable them.

With that, hell you could include this package (mainly meaning the core here) for comments if you wanted to (categories become like posts/user profiles, threads for each post or user profile and such). Weird Idea and not the best way to do it but would be interesting if this could become adaptable enough for such things.

**Comment by Riari** [comment]
Somewhat addressed in the latest release, [1.0.4](https://github.com/Riari/laravel-forum-frontend/releases/tag/1.0.4). It's not perfect, but something that I'll improve in laravel-forum 4.0/frontend 2.0. :)
